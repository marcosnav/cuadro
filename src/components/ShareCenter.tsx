import React, { FC } from 'react';
import styled from 'styled-components';
import { getShareFunc, SharePlatform } from './../utils';
import PrettyLink from './PrettyLink';

const Section = styled.section`
  margin-top: 42px;
  text-align: center;
`;

const Title = styled.h4`
  color: ${(props) => props.theme.TEXT};
  margin: 0;
  text-align: center;
  text-transform: uppercase;
`;

const ShareSites = styled.div`
  display: flex;
  font-size: 0.83em;
  margin-top: 10px;
  justify-content: center;
  width: 300px;

  & > * {
    margin: 0 10px;
  }
`;

const ShareCenter: FC = () => {
  return (
    <Section>
      <Title>
        Share The Joy
      </Title>
      <ShareSites>
        <PrettyLink onClick={getShareFunc(SharePlatform.FACEBOOK)}>
          Facebook
        </PrettyLink>
        <PrettyLink onClick={getShareFunc(SharePlatform.TWITTER)}>
          Twitter
        </PrettyLink>
      </ShareSites>
    </Section>
  );
};

export default ShareCenter;
