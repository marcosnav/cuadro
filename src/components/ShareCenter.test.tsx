import { mount } from 'enzyme';
import 'jest-enzyme';
import 'jest-styled-components';
import React from 'react';
import ShareCenter from './ShareCenter';

describe('<ShareCenter/>', () => {
  it('renders properly', () => {
    const component = mount(<ShareCenter/>);
    expect(component).toMatchSnapshot();
  });
});
