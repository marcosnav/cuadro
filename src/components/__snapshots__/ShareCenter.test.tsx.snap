// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShareCenter/> renders properly 1`] = `
.c3 {
  font-weight: bold;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-transition: opacity 0.3s ease;
  transition: opacity 0.3s ease;
}

.c3:hover {
  opacity: 0.4;
}

.c0 {
  margin-top: 42px;
  text-align: center;
}

.c1 {
  margin: 0;
  text-align: center;
  text-transform: uppercase;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.83em;
  margin-top: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 300px;
}

.c2 > * {
  margin: 0 10px;
}

<ShareCenter>
  <styled.section>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  margin-top: 42px;
  text-align: center;
",
            ],
          },
          "displayName": "styled.section",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "section",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <section
        className="c0"
      >
        <styled.h4>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  color: ",
                    [Function],
                    ";
  margin: 0;
  text-align: center;
  text-transform: uppercase;
",
                  ],
                },
                "displayName": "styled.h4",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "h4",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <h4
              className="c1"
            >
              Share The Joy
            </h4>
          </StyledComponent>
        </styled.h4>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": true,
                  "lastClassName": "c2",
                  "rules": Array [
                    "
  display: flex;
  font-size: 0.83em;
  margin-top: 10px;
  justify-content: center;
  width: 300px;

  & > * {
    margin: 0 10px;
  }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="c2"
            >
              <PrettyLink
                onClick={[Function]}
              >
                <styled.a
                  onClick={[Function]}
                  target="_blank"
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bdVaJa",
                          "isStatic": false,
                          "lastClassName": "c3",
                          "rules": Array [
                            "
  color: ",
                            [Function],
                            ";
  font-weight: bold;
  text-decoration: underline;
  transition: opacity 0.3s ease;

  &:hover {
    opacity: 0.4;
  }
",
                          ],
                        },
                        "displayName": "styled.a",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bdVaJa",
                        "target": "a",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    onClick={[Function]}
                    target="_blank"
                  >
                    <a
                      className="c3"
                      onClick={[Function]}
                      target="_blank"
                    >
                      Facebook
                    </a>
                  </StyledComponent>
                </styled.a>
              </PrettyLink>
              <PrettyLink
                onClick={[Function]}
              >
                <styled.a
                  onClick={[Function]}
                  target="_blank"
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bdVaJa",
                          "isStatic": false,
                          "lastClassName": "c3",
                          "rules": Array [
                            "
  color: ",
                            [Function],
                            ";
  font-weight: bold;
  text-decoration: underline;
  transition: opacity 0.3s ease;

  &:hover {
    opacity: 0.4;
  }
",
                          ],
                        },
                        "displayName": "styled.a",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bdVaJa",
                        "target": "a",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    onClick={[Function]}
                    target="_blank"
                  >
                    <a
                      className="c3"
                      onClick={[Function]}
                      target="_blank"
                    >
                      Twitter
                    </a>
                  </StyledComponent>
                </styled.a>
              </PrettyLink>
            </div>
          </StyledComponent>
        </styled.div>
      </section>
    </StyledComponent>
  </styled.section>
</ShareCenter>
`;
